ang/civicaseextras/CaseOutcome.js

```diff
@@ -0,0 +1,113 @@
+(function (_, angular, activityTypes) {
+  var module = angular.module('civicaseextras');
+
+  module.directive('civicaseExtrasCaseOutcome', function () {
+    return {
+      require: '^civicaseMasonryGrid',
+      scope: { case: '=' },
+      replace: true,
+      templateUrl: '~/civicaseextras/CaseOutcome.html',
+      controller: 'civicaseExtrasCaseOutcomeController',
+      link: civicaseExtrasCaseOutcomeLink
+    };
+
+    function civicaseExtrasCaseOutcomeLink ($scope, $element, attrs, masonryGrid) {
+      (function init () {
+        $scope.$on('civicaseExtrasCaseOutcome::loaded', oncaseOutcomeLoaded);
+      })();
+
+      function oncaseOutcomeLoaded () {
+        if ($scope.activityOutcomes.length) {
+          masonryGrid.addGridItemAt($element, 0);
+          $element.show();
```

Comment:
shouldn't this simply be a `ng-if` directive in the directive's template? After all you are deciding to display or not the element based on the current data, that's what angular is for

if you can get rid of this `.show()`/`.hide()`, then you don't need this link function at all, given that adding/removing a grid item should be done by the controller